gpkitmodels/GP/aircraft/tail/tail_boom.py

```diff
@@ -130,52 +132,54 @@ def setup(self, tailboom, htail, state):
         S = self.S = htail.planform.S
         E = self.E = tailboom.material.E
         Sy = self.Sy = tailboom.Sy
-        rhosl = self.rhosl = state.rhosl
-        Vne = self.Vne = state.Vne
+        qne = self.qne = state.qne
         CLmax = htail.planform.CLmax
         deta = tailboom.deta
         sigma = tailboom.material.sigma
 
-        return beam, [beam["dx"] == deta,
-                      F >= 0.5*rhosl*Vne**2*S,
-                      beam["\\bar{EI}"] <= E*I/F/l**2/2,
-                      Mr >= beam["\\bar{M}"][:-1]*F*l,
-                      sigma >= Mr/Sy,
-                      th == beam["\\theta"][-1],
-                      beam["\\bar{\\delta}"][-1]*CLmax <= kappa
-                     ]
-
-def makeTailBoom(N=2, tailboomSpar=TubeSpar):
-    class TailBoom(tailboomSpar):
-        """ Tail Boom Model
-
-        Variables
-        ---------
-        l                           [ft]        tail boom length
-        S                           [ft^2]      tail boom surface area
-        deta          1./(N-1)      [-]         normalized segment length
-
-        """
-
-        flight_model = TailBoomAero
-        tailLoad = TailBoomBending
-
-        def setup(self, N=N):
-            # exec parse_variables(TailBoom.__doc__)
-            self.N = N
-
-            l = self.l = Variable("l", "ft", "tail boom length")
-            S = self.S = Variable("S", "ft^2", "tail boom surface area")
-            self.deta = Variable("deta", 1./(N-1), "-",
-                                 "non-dim segment length")
-            b = self.b = Variable("b", "ft", "twice tail boom length")
-            self.cave = VectorVariable(N-1, "cave", "in",
-                                       "average segment width")
-            self.tau = Variable("tau", 1.0, "-", "thickness to width ratio")
-
-            self.spar = tailboomSpar.setup(self, N, self)
-
-            d0 = self.d0 = self.d[0]
-
-            return self.spar, [S == l*pi*d0, b == 2*l]
-    return TailBoom()
+        constraints = [beam["dx"] == deta,
+                       F >= qne*S,
+                       beam["\\bar{EI}"] <= E*I/F/l**2/2,
+                       Mr >= beam["\\bar{M}"][:-1]*F*l,
+                       sigma >= Mr/Sy,
+                       th == beam["\\theta"][-1],
+                       beam["\\bar{\\delta}"][-1]*CLmax*Nsafety <= kappa]
+
+        if hasattr(tailboom, "J"):
+            constraints.append(tailboom.J >= 1e-50*units("m^4"))
+
+        return constraints, beam
+
+class TailBoom(TubeSpar):
```

Comment:
since it uses `super` below, I wonder you could make a TailBoom with a different spar just by changing its parent? that's a slightly weird thing to ask the inheritance system to do I guess

Reply:
That is actually exactly what I do in `tail_tests.py`.  I was cautioned heavily on overstack to not do this, but I thought it was preferable so that I could keep the docstring parsing and allow me to change the `flight_model`, `tailLoad` and `secondaryWeight` properties.